src/dmd/dinterpret.d

```diff
@@ -6095,6 +6095,30 @@ public:
             se = (cast(ClassReferenceExp)ex).value;
             i = (cast(ClassReferenceExp)ex).findFieldIndexByName(v);
         }
+        else if (ex.op == TOK.typeid_)
+        {
+            auto tie = cast(TypeidExp) ex;
+            if (v.ident == Identifier.idPool("name"))
+            {
+                if (auto t = isType(tie.obj))
```

Comment:
`if (auto t = isType((cast(TypeidExp)ex).obj))`

This is the only use of `tie`, so there is no point in declaring a variable.

Reply:
it's in line with how the rest of the code in here does it.
`also tie` will be used more in the future, as I intend to extend  the CTFE typeid support.

Reply:
The review is based on the provided code, not on promises of future commits. When `tie` will be used more than once only then using a variable for it is going to be justified, but for now -1 line of code is the preferred way.

Reply:
There is one more reason to do it that way.
 `if (ex.op == TOK.typeid_)`
`cast(TypeIdExp)` should be close together since the former makes the later a valid operation.
These are extern(C++) classes which means that dynamic casts are not available.

Reply:
This is a valid argument only if `ex` is modified to point to a different type of class, but that is not happening here and it seems unlikely to happen because then you would rewrite the AST.

I am sure there a lot of situations where it makes sense to the the cast outside of the if branch, but the current situation is not one of them. For now, apply the required changes and when you will make the next PR extending the current code you can add the necessary modifications.

Reply:
BTW you could also use `isTypeidExp` instead of explicitly checking `op`

Reply:
which calls through the vtbl. why would you do that when you already had the op field?

Reply:
> which calls through the vtbl.

[Not since 2018](https://github.com/dlang/dmd/commit/edc7af85a29f67a6b5863e8338716beac03a9154)

Reply:
ah yeah ... I don't usually develop on this version of dmd :)